utils/property-propagation.stanza

```diff
@@ -0,0 +1,220 @@
+              ; the pin is connected to a net we haven't visited yet.
+              add(net-stack, n) when not visited?[n]
+
+; same as above with default updater
+public defn propagate-property-along-nets (root-module:JITXObject,
+                                           property-name:Symbol,
+                                           handle-instance: (JITXObject) -> Seqable<JITXObject>,
+                                           handle-conflict:(Seqable<jitx/client/JITXValue>) -> jitx/client/JITXValue) :
+  propagate-property-along-nets(root-module, property-name, handle-instance, handle-conflict, default-updater)
+
+; same as above with default handlers
+public defn propagate-property-along-nets (root-module:JITXObject,
+                                           property-name:Symbol) :
+  propagate-property-along-nets(root-module, 
+                                property-name, 
+                                default-instance-handler, 
+                                default-conflict-handler)
+
+; WIP: new rail-voltage propagation based on a two-pass approach
+; 
+; Example: 
+;   Here we propagate the "power-supply-pin" property along nets
+;   connected to any pin marked with this property, but map the 
+;   maximum voltage of the psu-pin to a property called "rail-voltage."
+; 
+;   All we need to do is call the main driver and define our handlers, 
+;   in this case we can use the default-instance-handler (which does 
+;   not propagate across components), with a custom conflict handler 
+;   and custom updater.
+;
+public defn propagate-rail-voltage-2 (root-module:JITXObject) : 
+  propagate-property-along-nets(root-module,
+                                `power-supply-pin,
+                                default-instance-handler,
+                                handle-conflict,
+                                updater) where :
+    defn handle-conflict (xs) : 
+      maximum $ 
+        for psp in filter-by<PowerSupplyPin>(xs) seq : 
+          match(supply-voltage(psp)) : 
+            (d:Double)     : d
+            (t:Toleranced) : max-value(t)
+
+    defn updater (p:JITXObject, name:Symbol, value:jitx/client/JITXValue) : 
+      set-property(p, `rail-voltage, value)
+    
+;==============================================================================
+;============================= Helpers ========================================
+;==============================================================================
+; TODO: refactor not to use undocumented internal APIs
+
+doc:"Returns if an object has a property."
+defn has-property? (obj:JITXObject, prop:Symbol) -> True|False : 
+  ; TODO: introspection instead of using internal APIs explicitly
+  jitx/client/get-property?(jitx/commands/JITX-CLIENT, false, obj, prop) is One
+
+doc:"Returns if a pin is connected to a component marked DNP"
+defn dnp-pin? (pin:JITXObject) -> True|False : 
```

Comment:
Use `ocdb/checks dnp?` to check if containing instance is DNP

Reply:
This would be good to do because we need to always filter DNP components, but it makes DNP a critical property that needs to be valid in the design (if used, that is). I think that the whole notion of DNP needs to be promoted to a more basic construct than a ephemeral property.

Reply:
Agreed. There's actually a good bit of nuance to this feature that I think it needs to be moved eventually into a core pass in the JITX compiler.

Reply:
Yes, that would be good. Is it too much to do now?

Reply:
Yes, it requires a more flushed out design. It would be helpful to have this pass at the user level for now to identify the intended behaviors before building out something internally.

Reply:
Ok, makes sense.